src/test/java/ay2122s1_cs2103t_w16_2/btbb/model/client/predicate/ClientPredicateCollectionTest.java

```diff
@@ -0,0 +1,89 @@
+package ay2122s1_cs2103t_w16_2.btbb.model.client.predicate;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import org.junit.jupiter.api.Test;
+
+import ay2122s1_cs2103t_w16_2.btbb.model.client.Client;
+import ay2122s1_cs2103t_w16_2.btbb.testutil.ClientBuilder;
+
+public class ClientPredicateCollectionTest {
+    private void addPredicates(ClientPredicateCollection clientPredicateCollection,
+                               List<Predicate<Client>> clientPredicates) {
+        for (Predicate<Client> clientPredicate : clientPredicates) {
+            clientPredicateCollection.addClientPredicate(clientPredicate);
+        }
+    }
+
+    @Test
+    public void equals() {
+        NameContainsKeywordsPredicate nameContainsKeywordsPredicate =
+                new NameContainsKeywordsPredicate(List.of("Alice", "Bob"));
+        PhoneContainsKeywordsPredicate phoneContainsKeywordsPredicate =
+                new PhoneContainsKeywordsPredicate(List.of("9427", "3217"));
+        AddressContainsKeywordsPredicate addressContainsKeywordsPredicate =
+                new AddressContainsKeywordsPredicate(List.of("Yishun", "Geylang"));
+        EmailContainsKeywordsPredicate emailContainsKeywordsPredicate =
+                new EmailContainsKeywordsPredicate(List.of("alice@gmail.com", "bob@gmail.com"));
+        ClientPredicateCollection clientPredicateCollection = new ClientPredicateCollection();
+        addPredicates(clientPredicateCollection, List.of(nameContainsKeywordsPredicate,
+                phoneContainsKeywordsPredicate, addressContainsKeywordsPredicate, emailContainsKeywordsPredicate));
+        ClientPredicateCollection diffOrderClientPredicateCollection = new ClientPredicateCollection();
+        addPredicates(diffOrderClientPredicateCollection, List.of(phoneContainsKeywordsPredicate,
+                addressContainsKeywordsPredicate, nameContainsKeywordsPredicate, emailContainsKeywordsPredicate));
+
+        // same order - must be equal
+        assertEquals(clientPredicateCollection, clientPredicateCollection);
+        // diff order - must be equal
+        assertEquals(clientPredicateCollection, diffOrderClientPredicateCollection);
+    }
+
+    @Test
+    public void notEquals() {
```

Comment:
All the test cases here should be moved to equals. You are testing the equal method.